types/types.go

```diff
@@ -184,6 +186,19 @@ type Metrics struct {
 	SizeMemoryCache uint64
 }
 
+type PerModuleMetrics struct {
+	Hits uint32 `msgpack:"hits"`
+	Size uint64 `msgpack:"size"`
+}
+
+type PinnedMetrics struct {
+	PerModule map[string]PerModuleMetrics `msgpack:"per_module"`
```

Comment:
I am happy either way, just some context:

In Go, those maps are always randomized, leading to nondeterministic iteration order. This caused determinism problems in Go based blockchains many times before. This is why people generally avoid maps as a safety measure. But here we don't have deterministic data anyways, so it's probably good to use that type.

What we could do as well is a BTreeMap in Rust and then turn this into a vector for FFI to get deterministic iteration order. But then again, here it does not matter really.

Reply:
I knew about Go's randomization of maps, it just didn't seem too important here.  
But we can totally attempt to use something like a tuple vector.

(Just needs some more trial and error on my side to figure out how to represent it)

Reply:
Got rid of the map